worker/instancemutater/manifold.go

```diff
@@ -53,17 +53,25 @@ func (config ManifoldConfig) Validate() error {
 	return nil
 }
 
-func (config ManifoldConfig) newWorker(environ environs.Environ, apiCaller base.APICaller, agent agent.Agent) (worker.Worker, error) {
+func (config ModelManifoldConfig) newWorker(environ environs.Environ, apiCaller base.APICaller, agent agent.Agent) (worker.Worker, error) {
 	if err := config.Validate(); err != nil {
 		return nil, errors.Trace(err)
 	}
-
+	// If we don't have a LXDProfiler, we should uninstall the worker as quickly
+	// as possible.
+	broker, ok := environ.(environs.LXDProfiler)
+	if !ok {
+		// If we don't have an LXDProfiler broker, there is no need to
+		// run this worker.
+		config.Logger.Debugf("Uninstalling worker because the broker is not a LXDProfiler %T", environ)
+		return nil, dependency.ErrUninstall
```

Comment:
I don't think ErrUninstall is correct, we should confirm. I don't think this is "stop running the worker", I think it is "uninstall jujud from this machine".

Reply:
This is probably what we want. https://github.com/juju/worker/blob/v1/dependency/interface.go#L126-L129